api/ombpdf/management/commands/import_pdfs_from_policies.py

```diff
@@ -0,0 +1,77 @@
+import logging
+from os.path import basename
+from typing import List, Optional, Set, Tuple
+
+import requests
+from django.core.management.base import BaseCommand
+from django.db.models import Exists, OuterRef
+from requests.exceptions import RequestException
+
+from document.models import DocNode
+from ombpdf.management.commands.scrape_memoranda import Url, parse_pdf
+from reqs.models import Policy
+
+OLD_DOMAIN = 'www.whitehouse.gov'
+NEW_DOMAIN = 'obamawhitehouse.archives.gov'
+PROBLEM_PDFS = (
+    'a11_2016.pdf',
+)
+logger = logging.getLogger(__name__)
+
+
+def best_url(urls: List[Url]) -> Optional[Url]:
+    """Try hitting the requested url; if it fails, fall back to an
+    alternate."""
+    for url in urls:
+        if basename(url) in PROBLEM_PDFS:
+            continue
+        try:
+            response = requests.head(url)
+            if response.status_code == 200:
+                return url
+        except RequestException:
+            continue
+    return None
+
+
+def import_from_policy(policy: Policy) -> bool:
+    """Convert a policy to a list of potential urls to try, pick the first
+    that works, then attempt to parse."""
+    uri = policy.uri.strip()
+    urls_to_try = [Url(uri)]
+    if OLD_DOMAIN in uri:
+        urls_to_try.append(Url(uri.replace(OLD_DOMAIN, NEW_DOMAIN)))
+    if policy.document_source:
+        urls_to_try.insert(0, Url(policy.document_source.url))
+
+    url = best_url(urls_to_try)
+    return bool(url and parse_pdf(policy, url))
+
+
+def import_pdfs_from_policies() -> Tuple[Set[str], Set[str]]:
+    """Attempt to import document text from all pdf policies which haven't
+    already been imported."""
+    successes, failures = set(), set()
+    has_docnodes = Exists(DocNode.objects.filter(policy=OuterRef('pk')))
+    query = Policy.objects \
+        .annotate(has_docnodes=has_docnodes) \
+        .filter(uri__endswith='.pdf') \
+        .filter(has_docnodes=False)     # not replacing data
```

Comment:
I noticed that there's some similar code in `scrape_memoranda()` from #801 that is almost identical, save for the middle `filter()`.  Do you think we'd use this query elsewhere? I actually recall pair programming #661 with you, in which we used the same `Exists(DocNode.objects.filter(policy=OuterRef('pk')))` annotation, except we called it `has_docnode` (singular) there instead of plural... I wonder if it'd be useful to somehow pull this out into a bit of reusable code, like a method on `Policy`'s `QuerySet` or manager class or something?

Eh, it's up to you really.

Reply:
I'll make a little helper function; I agree that it makes sense to be placed on `Policy`, but that's create a circular import.

Reply:
I'll add in an addendum PR as it'd probably be accurate to use the plural everywhere, but I don't want to modify the frontend here.